### ISSUE TITLE
Not compatible with Android Gradle plugin 3.6

### ISSUE THREAD

**Issue by mxalbert1996** [DeployGate/gradle-deploygate-plugin#90]
Got the following error with Android Gradle plugin 3.6.0-beta01.

```
Caused by: org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'property(interface org.gradle.api.file.Directory, property(interface org.gradle.api.file.Directory, fixed(class org.gradle.api.internal.file.DefaultFilePropertyFactory$FixedDirectory, /path/to/project/app/build/outputs/apk/debug)))' with class 'org.gradle.api.internal.file.DefaultFilePropertyFactory$DefaultDirectoryVar' to class 'java.io.File'
        at com.deploygate.gradle.plugins.artifacts.PackageAppTaskCompat.getApkInfo(PackageAppTaskCompat.groovy:17)
        at com.deploygate.gradle.plugins.artifacts.PackageAppTaskCompat$getApkInfo.call(Unknown Source)
        at com.deploygate.gradle.plugins.tasks.factory.AGPBasedUploadApkTaskFactory$_registerUploadApkTask_closure1.doCall(AGPBasedUploadApkTaskFactory.groovy:36)
        at com.sun.proxy.$Proxy82.execute(Unknown Source)
        at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:42)
        at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:42)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1$1.run(DefaultCollectionCallbackActionDecorator.java:100)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.reapply(DefaultUserCodeApplicationContext.java:60)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1.run(DefaultCollectionCallbackActionDecorator.java:97)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction.execute(DefaultCollectionCallbackActionDecorator.java:94)
        at org.gradle.internal.ImmutableActionSet$SetWithManyActions.execute(ImmutableActionSet.java:329)
        at org.gradle.api.internal.DefaultDomainObjectCollection.doAdd(DefaultDomainObjectCollection.java:264)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.doAdd(DefaultNamedDomainObjectCollection.java:112)
        at org.gradle.api.internal.DefaultDomainObjectCollection.add(DefaultDomainObjectCollection.java:258)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection$AbstractDomainObjectCreatingProvider.tryCreate(DefaultNamedDomainObjectCollection.java:942)
        ... 117 more
```

**Comment by jmatsu** [comment]
Thank you for your report.
It sounds AndroidStudio 3.6 may have changed the signature of some classes. Let me investigate it but it may take a little while.

**Comment by jmatsu** [comment]
@mxalbert1996 Could you please let me know the Gradle version?

**Comment by jmatsu** [comment]
Replicated with AGP 3.6.0-beta01 and Gradle 5.6.1. It's due to AGP's signature changes. 

```
PackageApplication#outputDirectory has changed from File to Provider<Directory>
PackageApplication#signingConfig has changed from FileCollection to SigningConfigProvider
```

**Comment by jmatsu** [comment]
Could you please try a SNAPSHOT `as_3_6_beta-SNAPSHOT` with following the links below?

- https://github.com/DeployGate/gradle-deploygate-plugin#snapshot
- https://jitpack.io/#DeployGate/gradle-deploygate-plugin/as_3_6_beta-SNAPSHOT

**Comment by mxalbert1996** [comment]
@jmatsu There's no problem with `as_3_6_beta-SNAPSHOT`. Thanks for the fix.

**Comment by jmatsu** [comment]
The fixing has been shipped with version 2.1.0.
